lib/CPAN/Common/Index/LocalPackage.pm

```diff
@@ -44,27 +46,29 @@ sub BUILD {
 
 sub cached_package {
     my ($self) = @_;
-    my $package = path( $self->cache, path( $self->source )->basename );
+    my $package = File::Spec->catfile(
+        $self->cache, File::Basename::basename($self->source)
+    );
     $package =~ s/\.gz$//;
     $self->refresh_index unless -r $package;
     return $package;
 }
 
 sub refresh_index {
     my ($self) = @_;
-    my $source = path( $self->source );
+    my $source = $self->source;
     if ( $source =~ /\.gz$/ ) {
-        ( my $uncompressed = $source->basename ) =~ s/\.gz$//;
-        $uncompressed = path( $self->cache, $uncompressed );
-        if ( !-f $uncompressed or $source->stat->mtime > $uncompressed->stat->mtime ) {
+        ( my $uncompressed = File::Basename::basename($source) ) =~ s/\.gz$//;
+        $uncompressed = File::Spec->catfile( $self->cache, $uncompressed );
+        if ( !-f $uncompressed or (stat $source)[9] > (stat $uncompressed)[9] ) {
```

Comment:
I'd like to see File::stat used for readability. Or, create a constant "MTIME" that is equal to 9 and can be used as the array index.